packages/image_picker/image_picker/ios/Classes/FLTPHPickerSaveImageToPathOperation.m

```diff
@@ -82,51 +84,63 @@ - (void)start {
+                                }];
+      return;
+    }
+
     [self.result.itemProvider
         loadObjectOfClass:[UIImage class]
         completionHandler:^(__kindof id<NSItemProviderReading> _Nullable image,
                             NSError *_Nullable error) {
           if ([image isKindOfClass:[UIImage class]]) {
-            __block UIImage *localImage = image;
-            PHAsset *originalAsset =
-                [FLTImagePickerPhotoAssetUtil getAssetFromPHPickerResult:self.result];
-
-            if (self.maxWidth != (id)[NSNull null] || self.maxHeight != (id)[NSNull null]) {
-              localImage = [FLTImagePickerImageUtil scaledImage:localImage
-                                                       maxWidth:self.maxWidth
-                                                      maxHeight:self.maxHeight
-                                            isMetadataAvailable:originalAsset != nil];
-            }
-            __block NSString *savedPath;
-            if (!originalAsset) {
-              // Image picked without an original asset (e.g. User pick image without permission)
-              savedPath =
-                  [FLTImagePickerPhotoAssetUtil saveImageWithPickerInfo:nil
-                                                                  image:localImage
-                                                           imageQuality:self.desiredImageQuality];
-              [self completeOperationWithPath:savedPath];
-            } else {
-              [[PHImageManager defaultManager]
-                  requestImageDataForAsset:originalAsset
-                                   options:nil
-                             resultHandler:^(
-                                 NSData *_Nullable imageData, NSString *_Nullable dataUTI,
-                                 UIImageOrientation orientation, NSDictionary *_Nullable info) {
-                               // maxWidth and maxHeight are used only for GIF images.
-                               savedPath = [FLTImagePickerPhotoAssetUtil
-                                   saveImageWithOriginalImageData:imageData
-                                                            image:localImage
-                                                         maxWidth:self.maxWidth
-                                                        maxHeight:self.maxHeight
-                                                     imageQuality:self.desiredImageQuality];
-                               [self completeOperationWithPath:savedPath];
-                             }];
-            }
+            [self processImage:image];
           }
         }];
   } else {
     [self setFinished:YES];
   }
 }
 
+- (void)processImage:(UIImage *)localImage API_AVAILABLE(ios(14)) {
+  PHAsset *originalAsset = [FLTImagePickerPhotoAssetUtil getAssetFromPHPickerResult:self.result];
+
+  if (self.maxWidth != (id)[NSNull null] || self.maxHeight != (id)[NSNull null]) {
```

Comment:
I know it's pre-existing code, but are these casts actually necessary?

Reply:
I don't want to touch this code because it works. But I have analysed the flow.  

1. The flutter side initialises the call.
```dart
Future<String?> _getImagePath({
    required ImageSource source,
    double? maxWidth,
    double? maxHeight,
    int? imageQuality,
    CameraDevice preferredCameraDevice = CameraDevice.rear,
  }) {
/// ....
return _channel.invokeMethod<String>(
      'pickImage',
      <String, dynamic>{
        'source': source.index,
        'maxWidth': maxWidth,
        'maxHeight': maxHeight,
        'imageQuality': imageQuality,
        'cameraDevice': preferredCameraDevice.index
      },
    );
}
```
2. Base on the information of channel mechanism. Dart nil will be convert to objective-c nil or NSNull if it's nested ([link](https://docs.flutter.dev/development/platform-integration/platform-channels?tab=type-mappings-obj-c-tab))
3. The plugin use just parses those arguments.
```objective-c
NSNumber *maxWidth = [_arguments objectForKey:@"maxWidth"];
NSNumber *maxHeight = [_arguments objectForKey:@"maxHeight"];
```

Does it mean that `maxWidth` and `maxHeight` are nil? If yes then the code should crash because we some nested method will try to access to the value. And as far as I know [NSNull null] is not equal to nil.

Reply:
My question wasn't whether the checks were necessary, it was whether the *casts* (to `id`) were necessary.

Reply:
Oh, I think is not necessary.
https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/NumbersandValues/Articles/Null.html

Reply:
Either there's a compiler warning without the casts, or there isn't; it's not an opinion-based question. If there isn't, we should remove them.

(The real problem is that `NSNull` should never be assigned to `NSNumber` variables in the first place, since it's a type violation; once that's fixed these would be `nil` checks instead. @cyanglaz Perhaps you could fix that in a small separate PR?)

Reply:
I fixed the use of `NSNull` in this plugin, so these check should be nil checks now (as they are in the current version of the code you are moving).